source/MRMesh/MRMarchingCubes.cpp

```diff
@@ -431,12 +432,13 @@ Expected<TriMesh> volumeToMesh( const V& volume, const MarchingCubesParams& para
                     SeparationPointSet set;
                     bool atLeastOneOk = false;
                     const float value = cache ? cache->get( loc ) : acc.get( loc );
-                    if ( nanChecker( value ) )
+                    const bool lower = value < params.iso;
+                    const bool notLower = value >= params.iso;
+                    if ( !lower && !notLower )
```

Comment:
mb create inline function with name indicating that is check for nan? or at least add comment that is non check